src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,127 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+interface Props {
+  onAdd: (newMovie: Movie) => void;
+}
+
+export const NewMovie = ({ onAdd }: Props) => {
+  const [movie, setMovie] = useState({
+    imdbId: '',
+
+    imdbUrl: '',
+
+    imgUrl: '',
+
+    description: '',
+
+    count: 0,
+
+    title: '',
+  });
+
+  const isFormValid = (): boolean =>
+    [
+      movie.title.trim(),
+      movie.imgUrl.trim(),
+      movie.imdbUrl.trim(),
+      movie.imdbId.trim(),
+    ].every(Boolean);
+
+  const isValid = isFormValid();
+
+  const resetForm = () => {
+    setMovie(prevSt => ({
+      imdbId: '',
+      imdbUrl: '',
+      imgUrl: '',
+      description: '',
+      count: prevSt.count + 1,
+      title: '',
+    }));
```

Comment:
```suggestion
    setMovie(DEFAULT_MOVIE);
    setCount(prev => prev + 1);
```
Create a default movie object to avoid repeating it